Distinguish between password fields

Brett208 opened the pull request:
Closes #54

DanRStevens commented on client/OPUNetTransportLayer.cpp:
```diff
@@ -1146,7 +1146,7 @@ bool OPUNetTransportLayer::DoImmediateProcessing(Packet &packet, sockaddr_in &fr
 			}
-			// Verify password
+			// Verify joinRequestPassword
 			if (strncmp(tlMessage.searchQuery.password, password, sizeof(password)) != 0) {
```
Is this part correct? It seems odd the comment was updated, but the field names were not.

DanRStevens commented on client/OPUNetTransportLayer.cpp:
```diff
@@ -167,7 +167,7 @@ bool OPUNetTransportLayer::HostGame(USHORT port, const char* password, const cha
-	// Copy the game creator name and password
+	// Copy the game creator name and joinRequestPassword
 	strncpy_s(hostedGameInfo.createGameInfo.gameCreatorName, creatorName, sizeof(hostedGameInfo.createGameInfo.gameCreatorName));
 	strncpy_s(this->password, password, sizeof(this->password));
```
Again, the comment was updated, but none of the field names.

Brett208 commented:
@DanRStevens, I think this commit addresses your comments and valid concerns...

 It looks like circle ci support has been removed from the project.

DanRStevens commented:
CircleCI was recently added, in preparation for the repository merge. The Server project uses CircleCI, however the Client project so far does not. (It's Windows only). The failures are because there is currently no CircleCI config file present.

I should perhaps either disable CircleCI for now, or maybe add a Mingw build.

----

I've review in detail in the morning.

Brett208 commented:
OK. I had assumed since there was a makefile already present that we had been compiling a circleci build in the past. I suppose I'm fine with merging things with failed circleci builds if it isn't established yet.
